cleantalk.php

```diff
@@ -1916,64 +1972,58 @@ function apbct_rc__update()
-        }
+        $result = activate_plugins($plugin);
 
-        $httpResponseCode = Helper::httpRequest(get_option('home'), array(), 'get_code');
+        apbct_update__outputResult(
+            'PLUGIN_ACTIVATING',
+            is_wp_error($result) || $result === false ? 'FAIL' : 'OK',
+            array('wp_error' => is_wp_error($result) ? $result->get_error_message() : '')
+        );
 
-        if ( strpos($httpResponseCode, '200') === false ) {
-            apbct_maintenance_mode__enable(30);
+        $httpResponseCode = Helper::httpRequestGetResponseCode(get_option('home'));
 
+        if ( $httpResponseCode != 200 ) {
             // Rollback
-            if ( version_compare(PHP_VERSION, '5.6.0') >= 0 && version_compare($wp_version, '5.3') >= 0 ) {
-                $rollback = new CleantalkUpgrader(
-                    new CleantalkUpgraderSkin(compact('title', 'nonce', 'url', 'plugin_slug', 'prev_version'))
-                );
-            } else {
-                $rollback = new CleantalkUpgrader(
-                    new CleantalkUpgraderSkinDeprecated(compact('title', 'nonce', 'url', 'plugin_slug', 'prev_version'))
+            apbct_maintenance_mode__enable(30);
+
+            $rollback = version_compare(PHP_VERSION, '5.6.0') >= 0 && version_compare($wp_version, '5.3') >= 0
+                ? new CleantalkUpgrader(new CleantalkUpgraderSkin(compact('title', 'nonce', 'url', 'plugin')))
+                : new CleantalkUpgrader(
+                    new CleantalkUpgraderSkinDeprecated(compact('title', 'nonce', 'url', 'plugin'))
                 );
-            }
             $rollback->rollback($plugin);
 
             apbct_maintenance_mode__disable();
 
-            // @todo add execution time
-
-            $response = array(
-                'error'           => 'BAD_HTTP_CODE',
-                'http_code'       => $httpResponseCode,
-                'output'          => substr(file_get_contents(get_option('home')), 0, 900),
-                'rollback_result' => $rollback->apbct_result,
+            apbct_update__outputResult(
+                'CHECK_RESPONSE',
+                'FAIL',
+                array(
+                    'error'           => 'BAD_HTTP_CODE',
+                    'http_code'       => $httpResponseCode,
+                    'output'          => htmlspecialchars(
+                        substr(Helper::httpRequestGetContent(get_option('home')), 0, 900)
+                    ),
+                    'rollback_result' => $rollback->apbct_result,
+                )
             );
-
-            die('FAIL ' . json_encode($response));
         }
 
-        $plugin_data = get_plugin_data(__FILE__);
-        $apbct_agent = 'wordpress-' . str_replace('.', '', $plugin_data['Version']);
-        ct_send_feedback('0:' . $apbct_agent);
-
-        die('OK ' . json_encode(array('agent' => $apbct_agent)));
+        apbct_update__outputResult(
+            'CHECK_RESPONSE',
+            'OK'
+        );
+    } else {
```

Comment:
Dead code. No need to process it here. Because the apbct_update__outputResult() on 1975 line will do it.

Reply:
You're right https://github.com/CleanTalk/wordpress-antispam/pull/96/commits/ab2170147f07b760b135ce476285335ce0bfeadb